dev/test/field-value.ts

```diff
@@ -121,6 +121,15 @@ describe('FieldValue.arrayUnion()', () => {
     });
   });
 
+  it('must not contain directly nested arrays', () => {
+    return createInstance().then(firestore => {
+      const docRef = firestore.doc('collectionId/documentId');
+      expect(() => docRef.set({foo: FieldValue.arrayUnion([])})).to.throw(
+        `Nested arrays are not supported`
```

Comment:
Can you modify the test to assert on the full error message (including 'invalidArgumentMessage'). I wanted to see what the full error message looks like, but I would rather let you do this research and convey the result via this test.

Reply:
done.